<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>异步函数的异常处理</title>
  </head>
  <body>
    <script>
      async function foo() {
        console.log('---1')
        console.log('---2')
        throw new Error('出错了')
        console.log('---3')
      }

      foo()
        .then((res) => {})
        .catch((err) => {
          console.log(err)
        })

      // ---1
      // ---2
      // Error: 出错了
    </script>
  </body>
</html>
